platform/security/servlet/security-servlet-session-expiry/src/main/java/org/codice/ddf/security/servlet/expiry/SessionManagementService.java

```diff
@@ -0,0 +1,156 @@
+            if (securityToken instanceof SecurityTokenHolder) {
+                List<SecurityAssertionImpl> values =
+                        ((SecurityTokenHolder) securityToken).getRealmTokenMap()
+                                .values()
+                                .stream()
+                                .map(SecurityAssertionImpl::new)
+                                .collect(Collectors.toList());
+                values.sort((o1, o2) -> {
+                    long l = o1.getNotOnOrAfter()
+                            .getTime() - o2.getNotOnOrAfter()
+                            .getTime();
+                    if (l > 0) {
+                        return 1;
+                    } else if (l < 0) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                });
+                timeLeft = values.isEmpty() ?
+                        0 :
+                        values.get(0)
+                                .getNotOnOrAfter()
+                                .getTime();
+            }
+        }
+        return Response.ok(new ByteArrayInputStream(Long.toString(Math.max(
+                timeLeft - clock.millis(), 0))
+                .getBytes(StandardCharsets.UTF_8)))
+                .build();
+    }
+
+    @GET
+    @Path("/renew")
+    public Response getRenewal(@Context HttpServletRequest request) {
+        HttpSession session = request.getSession(false);
+        final Response[] response = new Response[1];
+        if (session != null) {
+            Object securityToken = session.getAttribute(SecurityConstants.SAML_ASSERTION);
+            if (securityToken instanceof SecurityTokenHolder) {
+                SecurityTokenHolder tokenHolder = (SecurityTokenHolder) securityToken;
+                Map<String, SecurityToken> realmTokenMap = tokenHolder.getRealmTokenMap();
+                realmTokenMap.keySet()
+                        .forEach(s -> {
+                            try {
+                                doRenew(s, realmTokenMap.get(s), tokenHolder);
+                            } catch (SecurityServiceException e) {
+                                response[0] = Response.serverError()
+                                        .build();
+                                LOGGER.error("Failed to renew", e);
+                            }
+                        });
+            }
+        }
+        if (response[0] == null) {
+            response[0] = Response.ok()
+                    .build();
+        }
+        return response[0];
+    }
+
+    @GET
+    @Path("/invalidate")
+    public Response getInvalidate(@Context HttpServletRequest request) {
+        StringBuffer requestURL = request.getRequestURL();
+        String requestQueryString = request.getQueryString();
+        return Response.temporaryRedirect(URI.create(requestURL.substring(0,
+                requestURL.indexOf("services"))
```

Comment:
❗️ Should use `org.codice.ddf.system.rootContext` system property instead of hardcoding.